stub/src/main/java/io/grpc/stub/MetadataUtils.java

```diff
@@ -143,4 +148,42 @@ public void onClose(Status status, Metadata trailers) {
       }
     }
   }
+
+  /**
+   * Returns a ServerInterceptor that attaches a given set of headers to every response.
+   *
+   * @param extraHeaders the headers to be added to each response. Caller gives up ownership.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/11462")
+  public static ServerInterceptor newAttachHeadersServerInterceptor(Metadata extraHeaders) {
+    return new MetadataAttachingServerInterceptor(extraHeaders);
+  }
+
+  private static final class MetadataAttachingServerInterceptor implements ServerInterceptor {
+
+    private final Metadata extraHeaders;
+
+    MetadataAttachingServerInterceptor(Metadata extraHeaders) {
+      this.extraHeaders = extraHeaders;
+    }
+
+    @Override
+    public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
+        ServerCall<ReqT, RespT> call, Metadata headers, ServerCallHandler<ReqT, RespT> next) {
+      return next.startCall(new HeaderAttachingServerCall<>(call), headers);
+    }
+
+    final class HeaderAttachingServerCall<ReqT, RespT>
+        extends SimpleForwardingServerCall<ReqT, RespT> {
+      HeaderAttachingServerCall(ServerCall<ReqT, RespT> delegate) {
+        super(delegate);
+      }
+
+      @Override
+      public void sendHeaders(Metadata headers) {
```

Comment:
FYI: If the response is trailers-only (generally an error), this won't attach the user's headers.

Reply:
Fixed. Thanks for noticing.